### PR TITLE
 Major docstring fixes and reformats (Napoleon Style FTW!) 

### PR DESCRIPTION
**Pull request by ghcs27**
Lots of core.py docstrings, reformatted into Napoleon style.

### CODE REVIEW

**Comment by KennethNielsen** [comment]
Without reviewing this. I imaging most of this is non-problematic and **much** needed. The only exception if removing the "Raises SoCoException" from the docs, which I'm not sure I agree on, and which I think should be discussed as matter of policy.

But I would like to get all the rest of your work in right away, to prevent this rather large PR to go stale. So how about cherrypicking commit e81bd71 into a different branch (and PR) and then rebasing this one, without that commit?. That way we can continue with the large part here immediately.

**Comment by ghcs27** [comment]
Sorry, I messed up my Git files, so getting out this PR is not as easy as hoped. Will resolve that soon.

**Comment by KennethNielsen** [comment]
@ghcs27 let me know if you need help. Remember you can always reset your local version of this branch to the one on github, that this PR is created from, to start working from this starting point. In any case, let me know if you need help with git.

**Comment by ghcs27** [comment]
It will probably be faster to make a 'put raises back in' commit than to rebase the three following commits (lots of conflicts, some of them not easily resolvable).
Anyway, it's not that much work, I just have to find time to do it.

**Comment by ghcs27** [comment]
I just put the 'Raises SoCoException' lines back in, but I just can't come to terms with it.
It's only in some seemingly random docstrings, although it is true of all of them. The current string also not compliant with the Napoleon style.
To make the docstrings consistent, we'll have to either put a Napoleon version of it in *all* the docstrings or remove it.

File: soco/core.py [hunk 1]
```diff
@@ -686,7 +680,7 @@ def mute(self, mute):
 
     @property
     def volume(self):
-        """The speaker's volume.
+        """int: The speaker's volume.
 
         An integer between 0 and 100.
```
[the remark this answers is not in the document]
Comment (KennethNielsen) [reply soco/core.py]: if it fits in a line anyway

File: soco/core.py [hunk 2]
```diff
@@ -759,8 +753,9 @@ def treble(self, treble):
 
     @property
     def loudness(self):
-        """The Sonos speaker's loudness compensation. True if on, otherwise
-        False.
+        """bool: The Sonos speaker's loudness compensation.
+
+        True if on, False otherwise.
```
Comment (KennethNielsen) [inline soco/core.py]: this might also be merged into the top line

File: soco/core.py [hunk 3]
```diff
@@ -1113,9 +1088,7 @@ def is_playing_radio(self):
 
     @property
     def is_playing_line_in(self):
-        """ Is the speaker playing line-in?
-
-        return True or False
+        """bool: Is the speaker playing line-in?
         """
```
Comment (KennethNielsen) [inline soco/core.py]: no newline

File: soco/core.py [hunk 4]
```diff
@@ -1412,8 +1374,9 @@ def queue_size(self):
         return queue_size
 
     def get_sonos_playlists(self, *args, **kwargs):
-        """ Convenience method for:
-            get_music_library_information('sonos_playlists')
+        """ Convenience method for
```
Comment (KennethNielsen) [inline soco/core.py]: No space before Con...

File: soco/core.py [hunk 5]
```diff
@@ -1494,13 +1460,8 @@ def remove_from_queue(self, index):
         """ Remove a track from the queue by index. The index number is
```
Comment (KennethNielsen) [inline soco/core.py]: Let's loose the space before Remove here as well

File: soco/core.py [hunk 6]
```diff
@@ -1528,9 +1483,9 @@ def get_favorite_radio_shows(self, start=0, max_items=100):
         """Get favorite radio shows from Sonos' Radio app.
 
         Returns:
-        A list containing the total number of favorites, the number of
-        favorites returned, and the actual list of favorite radio shows,
-        represented as a dictionary with `title` and `uri` keys.
+            A list containing the total number of favorites, the number of
```
Comment (KennethNielsen) [inline soco/core.py]: I suspected that this was wrong, so I checked the code. That first "list" should be "dict". But shouldn't that return type be documented properly. I guess these have disappeared anyway with one of your other PR's

File: soco/core.py [hunk 7]
```diff
@@ -1707,9 +1649,10 @@ def remove_sonos_playlist(self, sonos_playlist):
     def add_item_to_sonos_playlist(self, queueable_item, sonos_playlist):
         """Adds a queueable item to a Sonos' playlist.
 
-        :param queueable_item: the item to add to the Sonos' playlist
-        :param sonos_playlist: the Sonos' playlist to which the item should
-            be added
+        Args:
+            queueable_item: the item to add to the Sonos' playlist
```
Comment (KennethNielsen) [inline soco/core.py]: I guess both of these have types that could be referenced properly

File: soco/core.py [hunk 8]
```diff
@@ -433,19 +470,14 @@ def ramp_to_volume(self, volume, ramp_type='SLEEP_TIMER_RAMP_TYPE'):
...
-        required as an argument, where the first index is 0.
-
-        index: the index of the track to play; first item in the queue is 0
-        start: If the item that has been set should start playing
-
-        Returns:
-        True if the Sonos speaker successfully started playing the track.
-        False if the track did not start (this may be because it was not
-        requested to start because "start=False")
+        """ Play a track from the queue by index.
```
Comment (KennethNielsen) [inline soco/core.py]: No space before Play

File: soco/core.py [hunk 9]
```diff
@@ -1513,13 +1474,7 @@ def remove_from_queue(self, index):
...
     @only_on_master
     def clear_queue(self):
-        """Removes all tracks from the queue.
-
-        Returns:
-        True if the Sonos speaker cleared the queue.
-
-        Raises SoCoException (or a subclass) upon errors.
-        """
+        """Removes all tracks from the queue."""
```
Comment (KennethNielsen) [inline soco/core.py]: It should probably be """Remove all..."""

**Comment by KennethNielsen** [comment]
Just finished the review. Just a bunch of nit-pick comments. Nice work.

**Comment by ghcs27** [comment]
Thanks for the review, seems like I made quite some formatting blunders.
The one thing I'm not sure I agree on is putting the value range into the first line. I feel this makes the first line less clear and concise. But if you feel strongly about this please let me know.

**Comment by KennethNielsen** [comment]
@ghcs27 no I don't feel strongly about the range in the first line, so if you think it is better your way, let's go with that.

File: soco/core.py [hunk 10]
```diff
@@ -1261,15 +1223,15 @@ def get_speaker_info(self, refresh=False, timeout=None):
...
+            timeout: How long to wait for the server to send
+                data before giving up, as a float, or a
+                `(connect timeout, read timeout)` tuple
+                e.g. (3, 5). Default is no timeout.
 
         Returns:
-        Information about the Sonos speaker, such as the UID, MAC Address, and
-        Zone Name.
+            dict: Information about the Sonos speaker, such as the UID,
+            MAC Address, and Zone Name.
```
Comment (KennethNielsen) [inline soco/core.py]: does that format correctly. I thought the second line "MAC Add...." should be indented.

File: soco/core.py [hunk 11]
```diff
@@ -1527,9 +1484,9 @@ def get_favorite_radio_shows(self, start=0, max_items=100):
         """Get favorite radio shows from Sonos' Radio app.
 
         Returns:
-        A list containing the total number of favorites, the number of
-        favorites returned, and the actual list of favorite radio shows,
-        represented as a dictionary with `title` and `uri` keys.
+            dict: A dictionary containing the total number of favorites, the
+            number of favorites returned, and the actual list of favorite radio
```
Comment (KennethNielsen) [inline soco/core.py]: Same here. But I might be mistaken. If you have the compiled docs from this branch at hand, maybe you can just check if I'm correct

**Comment by ghcs27** [comment]
It's actually correct like this: The 'Returns' section has to be the same indentation, it's a quirk of the napoleon convention. It took me a while to figure that out when I started...

**Comment by KennethNielsen** [comment]
Great.Thanks for the work.

Merging.

Remember to add a note to the release notes.

**Pull request merged**